### ISSUE TITLE
@Method decorator not working

### ISSUE THREAD

**Issue by 6button9** [ionic-team/stencil#276]
**Current behavior:**

@Method decorator does not appear to work, per example documents.

**Expected behavior:**

To be able to call the method, as per the example documents.

**Steps to reproduce:**

Attempted to compile code below.
if this line of code is commented out,
 // myElement.postToConsole('test');
all the console.logs appear in the console.
**Related code:**

```
// index.html
<!DOCTYPE html>
<html dir="ltr" lang="en">
<head>
  <meta charset="utf-8">
  <title>Stencil Starter App</title>

  <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0">
  <meta name="theme-color" content="#16161d">
  <meta name="apple-mobile-web-app-capable" content="yes">

  <script src="/build/app.js"></script>

  <link rel="apple-touch-icon" href="/assets/icon/icon.png">
  <link rel="icon" type="image/x-icon" href="/assets/icon/favicon.ico">
  <link rel="manifest" href="/manifest.json">
</head>
<body>

  <my-name first="John" last="Btn"></my-name>
  
  <script>
    const myElementH = document.querySelector('my-name');
    console.log(myElementH);
    //It does not work here either.
    myElementH.postToConsole('testH');
    console.log("afterH");
  </script>
</body>
</html>

//Stencil my-name.tsx
import { Component, Prop, Method } from '@stencil/core';

@Component({
  tag: 'my-name',
  styleUrl: 'my-name.scss'
})
export class MyName {

  @Prop() first: string;
  @Prop() last: string;

  @Method()
  postToConsole(text: string) {
    console.log(text);
  }

  render() {
    return (
      <div onClick={ _ => this.postToConsole('click')}>
        Hello, my name is {this.first} {this.last}
      </div>
    );
  }
}

const myElement = document.querySelector('my-name');
console.log(myElement);
//It does not work here.
myElement.postToConsole('test');
console.log("after");
```

**Other information:**

The remainder of the app is exactly as cloned from the starter app.

**Comment by adamdbradley** [comment]
I think what's happening here is that you're immediately calling the method before the component has finished loading. In your example `my-name` may have been connected, but it hasn't finished lazy loading the component yet. So I think this is a docs issue, and we need to explain better how to know when the app has finished loading. Thanks

**Comment by 6button9** [comment]
Adam,

Cool, I called it with a two second setTimeout and it worked, in the HTML
script tag.

How come, I cannot get the app to compile if I try to use the method
anywhere else in the code.

If I comment out the   //myElement.postToConsole('runElement'); function
call. The rest of the code will run and is callable by clicking on the
text, by calling the runElement function. I set up in a separate .tsx file
and export out the function. If I remove the comment tags, I get the
compiler error.


export const runElement = () =>{
  const myElement = document.querySelector('my-name');
  console.log(myElement);
  //myElement.postToConsole('runElement');
  console.log("after");

[ ERROR ]  typescript: src/components/method/methods.tsx, line: 4
           Property 'postToConsole' does not exist on type 'Element'.

      L3:  console.log(myElement);
      L4:  myElement.postToConsole('runElement');
      L5:  console.log("after");

[16:36.9]  build failed, watching for changes... in 2.55 s
